lib/mint/http1/headers.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Mint.HTTP1.Headers do
+  alias Mint.Core.Util
+
+  # format of headers is [{header_name, canonical_header_name, value}]
```

Comment:
Instead of having this as a comment, let's add it as a type so we can also use it for specs:

```suggestion
  @type header() :: {original_name :: String.t(), canonical_name :: String.t(), value :: String.t()}
  @type raw_header() :: {original_name :: String.t(), value :: String.t()}
```